tests/test_updater_ng.py

```diff
@@ -326,6 +333,69 @@ def test_non_existing_target_file(self) -> None:
         with self.assertRaises(exceptions.DownloadHTTPError):
             self.updater.download_target(info)
 
+    # TAP 14 tests ~ REMOVE COMMENT BEFORE THE NEXT COMMIT!!
+
+    # test case to check for the TAP 14 folder
+    def test_check_folder_tap14(self) -> None:
+        # Creating the parent folder for the TAP 14 folder
+        self.assertTrue(os.path.isdir(self.tap14_directory))
+
+    def test_check_tap14_contents(self) -> None:
+        # Checking specific files inside TAP 14
+        filenames = ["targets.json", "root.json"]
+        for file in filenames:
+            self.assertTrue(
+                os.path.isfile(os.path.join(self.tap14_directory, file))
+            )
+
+        # Checking specific folders inside TAP 14
+        foldernames = ["targets", "1.0.0"]
+        for folder in foldernames:
+            self.assertTrue(
+                os.path.isdir(os.path.join(self.tap14_directory, folder))
+            )
+
+    def test_get_spec_version1(self) -> None:
+        # This uses the default SUPPORTED_VERSIONS variable from updater.py
+        with self.assertRaises(exceptions.DownloadError):
```

Comment:
`assertRaises` takes an optional second argument: a message. This is really nice way to tell readers of your test code what you're checking:

`self.assertRaises(exceptions.DownloadError, "4 is not a supported version")`.

Go through and do that for every `assertRaises` or `assertEqual` call

Reply:
Fyi, `test_get_spec_version1` now `test_get_spec_version_supported` doesn't pass because `SUPPORTED_VERSIONS` isn't defined correctly.

Reply:
That's fine—in fact, really `test_get_spec_version` should only handle major version 1. So let's update the test cases